### PR TITLE
Replace Cheshire with Jsonista & ByteResponse

### LINKED ISSUE #62: Change default JSON formatter from Cheshire to Jsonista
The default JSON adapter could be Jsonista => the whole default lib would have explicit configuration. As this is a breaking change (all globally defined encoders vanish), it would require version `1.0.0`.

### PR DESCRIPTION
**Pull request by ikitommi**
* Cheshire -> Jsonista: fixes #62 
* Welcome `ByteResponse`s! Jsonista now 25% faster in blocking mode (e.g. `InputStreams`)

perf-json-relative2

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,27 @@
+## 0.6.0-SNAPSHOT
+
+* **BREAKING**: [Cheshire](https://github.com/dakrone/cheshire) in dropped in favor of [Jsonista](https://github.com/metosin/jsonista) as the default JSON formatter (faster, explicit configuration)
+  * `muuntaja.format.json` => `muuntaja.format.cheshire`
+  * `muuntaja.format.jsonista` => `muuntaja.format.json`
+  * The `muuntaja.format.json` formatter takes now jsonista options directly, with an asertion to fail fast if old options are used:
+     * `:key-fn` => `:encode-key-fn` and `:decode-key-fn`
+     * `:bigdecimals?` => `:bigdecimals`
+* move everyting from `muuntaja.records` into `muuntaja.core`
+* Encoders can now return `muuntaja.protocols/ByteResponse`, a wrapper for `byte-array`. It can be streamed effectively with both Ring (via `ring.protocols/StreamableResponseBody`) and via NIO-enables servers like [Aleph](https://github.com/ztellman/aleph) and [Immutant (perf fork)](https://github.com/ikitommi/immutant/pull/1)
```
Comment (miikka) [inline CHANGELOG.md]: [nit] Typo `NIO-enables` -> `NIO-enabled`

File: src/clj/muuntaja/protocols.clj [hunk 2]
```diff
@@ -2,7 +2,15 @@
   (:require [clojure.java.io :as io]
             [muuntaja.util :as util])
   (:import (clojure.lang IFn AFn)
-           (java.io ByteArrayOutputStream ByteArrayInputStream InputStreamReader BufferedReader InputStream Writer)))
+           (java.io ByteArrayOutputStream ByteArrayInputStream InputStreamReader BufferedReader InputStream Writer OutputStream)))
+
+(deftype ByteResponse [bytes]
+  IFn
```
Comment (miikka) [inline src/clj/muuntaja/protocols.clj]: I see that `StreamableResponse` below is callable/`IFn`, but honestly I don't remember why. When are you supposed to call these?

Comment (ikitommi) [reply src/clj/muuntaja/protocols.clj]: With an OutputStream, like `System/out`?Could be huge lazy dataset, but I guess `io/input-stream` already handles that. Maybe we should remove those `IFn`s?

File: CHANGELOG.md (hunk 1 above)
Comment (ikitommi) [reply CHANGELOG.md]: fixed

**Pull request merged**